docs/sources/next/javascript-api/k6-experimental/browser/elementhandle/setinputfiles.md

```diff
@@ -0,0 +1,72 @@
+| file.mimeType       | string      | `''`    | The type of the file content e.g. `text/plain`.                                                                                                                                                                                                                                                                                               |
+| file.buffer         | ArrayBuffer | `[]`    | base64 encoded content of the file.                                                                                                                                                                                                                                                                                                           |
+| options             | object      | `null`  | This is an optional parameter.                                                                                                                                                                                                                                                                                                                |
+| options.noWaitAfter | boolean     | `false` | If set to `true` and a navigation occurs from performing this action, it will not wait for it to complete.                                                                                                                                                                                                                                    |
+| options.timeout     | number      | `30000` | Maximum time in milliseconds. Pass `0` to disable the timeout. Default is overridden by the `setDefaultTimeout` option on [BrowserContext](https://grafana.com/docs/k6/<K6_VERSION>/javascript-api/k6-experimental/browser/browsercontext/) or [Page](https://grafana.com/docs/k6/<K6_VERSION>/javascript-api/k6-experimental/browser/page/). |
+
+### Example
+
+{{< code >}}
+
+```javascript
+import { browser } from 'k6/experimental/browser';
+import encoding from 'k6/encoding';
+
+export const options = {
+  scenarios: {
+    browser: {
+      executor: 'shared-iterations',
+      options: {
+        browser: {
+          type: 'chromium',
+        },
+      },
+    },
+  },
+};
+
+export default async function () {
+  const page = browser.newPage();
+
+  // In this example we create a simple web page with an upload input field.
+  page.setContent(`
+    <html>
+      <head></head>
+      <body>
+          <!-- Simple file upload form -->
+          <form method="POST" action="/upload" enctype="multipart/form-data">
+              <input type="file" id="upload" multiple />
+              <input type="submit" value="Send" />
+          </form>
+      </body>
+    </html>`);
+
+  const eh = page.$('input[id="upload"]');
+
+  // The file is set to the input element with the id "upload".
+  eh.setInputFiles({
```

Comment:
Where would this file have to be for this example to work?

Reply:
The file is inline in the code i.e. it's all there in the object (`name`, `mimetype` and `buffer`). This new API doesn't work with the local filesystem at all. A user will need to work with the experimental FS module to do that.